### DISCUSSION TITLE
[Python] How to pass a list as a cypher query parameter?

### DISCUSSION THREAD

**Discussion by rafsun42** [apache/age#550] [Q&A]
When I run this following query, I get the error `TypeError('not all arguments converted during string formatting')`.
```python
ag.execCypher(
        """
        MATCH (c)
        WHERE
            NOT (c.id IN []) AND
            (c.required = true OR c.id IN [%s])
        WITH c
            OPTIONAL MATCH (p)-[]->(c)
            RETURN c.id, collect(p.id)
        """,
        cols=["a TEXT", "b AGTYPE"],
        params=(['a', 'b', 'c'])
    )
```

However, if I insert the list with python's string formatter and `{..}` placeholder, it works. But that is unsafe.

I am not very familiar with psycopg2, but it looks like a missing feature in the age's python driver that didn't map to psycopg2 correctly. Can anyone confirm?

**Comment by moiz697** [comment]
you can use the join method to concatenate the list of strings into a single string, and then pass that string as the params argument. For e.g:-params=','.join(['a', 'b', 'c'])
